### PR TITLE
Adds random light flicker

### PR DESCRIPTION
**Pull request by tetraphobia**
:cl:
+Increases station spookiness level
/:cl:

### CODE REVIEW

**Comment by MrFroztee** [comment]
Let's christen this as our first PR in what must be seven years.
gnarlymusicalcat

**Comment by SheepFighter** [comment]
One would not think that so simple a change could crash Dream Daemon on start up, but here we are...

Time for me to do some tinkering...

**Comment by SheepFighter** [comment]
The good news is that I got it working.
The bad news is that the flickering happens with comical alacrity.
It's like a _Poltergeist_ parody.

Anyhow, I suspect this is going to need a universal timer, so as to stagger it. A random interval of 3-10 minutes between flickers would probably do nicely.

File: code/modules/power/lighting.dm [hunk 1]
```diff
@@ -395,6 +397,12 @@
 			update(0)
 		flickering = 0
 
+// random flicker
+/obj/machinery/light/process()
+	..()
```
Comment (SheepFighter) [inline code/modules/power/lighting.dm:402]: Bad idea, forgetting this.

File: code/modules/power/lighting.dm [hunk 2]
```diff
@@ -395,6 +397,12 @@
 			update(0)
 		flickering = 0
 
+// random flicker
+/obj/machinery/light/process()
+	..()
+	if(prob(3))
```
Comment (FlattestGuitar) [inline code/modules/power/lighting.dm]: A single `flicker()` call takes on average as much time as its argument - in this case it's 20 seconds.

So, every 2 seconds you have a 3% chance for each and every light on the station to start flickering for 20 seconds.

That means that every light at all times has a `1 - (97%)^10 ~= 10%` chance to be flickering in any given moment.

Tone that down to 1%, you don't want a rave, you want a space station.

**Comment by SheepFighter** [comment]
Sorry for not getting back sooner. The newest version is much better, but it's still far too frequent....
It might just be simpler to add a hard delay...

**Pull request closed without merging**